client/packages/common/src/hooks/useConfirmOnLeaving/useConfirmOnLeaving.ts

```diff
@@ -65,14 +64,18 @@ export const useBlockNavigation = () => {
     return false;
   });
 
+  const shouldBlockRefresh = blockers.some(
+    b => b.shouldBlock && !b.options?.allowRefresh
+  );
```

Comment:
Oop she got on that side quest grind - even after you saved the encounter, you weren't able to refresh because the "mark as visited on leaving" blocker was preventing it! So added option to prevent navigation but not refresh